hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/metrics/ContractMetrics.java

```diff
@@ -0,0 +1,233 @@
+    private Counter rejectedEthType3Counter;
+
+    private static final Map<HederaFunctionality, String> POSSIBLE_FAILING_TX_TYPES = Map.of(
+            CONTRACT_CALL, "contractCall", CONTRACT_CREATE, "createContract", ETHEREUM_TRANSACTION, "callEthereum");
+
+    private static final String METRIC_CATEGORY = "app";
+    private static final String METRIC_SERVICE = "SmartContractService";
+    private static final String METRIC_TXN_UNIT = "txs";
+
+    // Templates:  %1$s - HederaFunctionality name
+    //             %2$s - METRIC_SERVICE
+    //             %3$s - short specific metric description
+
+    private static final String REJECTED_NAME_TEMPLATE = "%2$s:Rejected-%1$s_total";
+    private static final String REJECTED_DESCR_TEMPLATE = "submitted %1$s %3$s rejected by pureChecks";
+
+    private static final String REJECTED_TXN_SHORT_DESCR = "txns";
+    private static final String REJECTED_TYPE3_SHORT_DESCR = "Ethereum Type 3 txns";
+    private static final String REJECTED_FOR_GAS_SHORT_DESCR = "txns with not even intrinsic gas";
+
+    private static final String REJECTED_TYPE3_FUNCTIONALITY = "ethType3BlobTransaction";
+
+    @Inject
+    public ContractMetrics(
+            @NonNull final Supplier<Metrics> metricsSupplier,
+            @NonNull final Supplier<ContractsConfig> contractsConfigSupplier) {
+        this.metricsSupplier = requireNonNull(
+                metricsSupplier, "metrics supplier (from platform via ServicesMain/Hedera must not be null");
+        this.contractsConfigSupplier =
+                requireNonNull(contractsConfigSupplier, "contracts configuration supplier must not be null");
+    }
+
+    public void createContractMetrics() {
+
+        final var contractsConfig = requireNonNull(contractsConfigSupplier.get());
+        this.p1MetricsEnabled = contractsConfig.metricsSmartContractPrimaryEnabled();
+        this.p2MetricsEnabled = contractsConfig.metricsSmartContractSecondaryEnabled();
+
+        final var metrics = requireNonNull(metricsSupplier.get());
+
+        if (p1MetricsEnabled) {
+            // Rejected transactions counters
+            for (final var txKind : POSSIBLE_FAILING_TX_TYPES.keySet()) {
+                final var name = toRejectedName(txKind, REJECTED_TXN_SHORT_DESCR);
+                final var descr = toRejectedDescr(txKind, REJECTED_TXN_SHORT_DESCR);
+                final var config = new Counter.Config(METRIC_CATEGORY, name)
+                        .withDescription(descr)
+                        .withUnit(METRIC_TXN_UNIT);
+                final var metric = newCounter(metrics, config);
+                rejectedTxsCounters.put(txKind, metric);
+            }
+
+            // Rejected transactions because they don't even have intrinsic gas
+            for (final var txKind : POSSIBLE_FAILING_TX_TYPES.keySet()) {
+                final var functionalityName = POSSIBLE_FAILING_TX_TYPES.get(txKind) + "DueToIntrinsicGas";
```

Comment:
In theory, this could be put into the for loop above.  I guess you broke this out for clarity?

Reply:
Yes, and, yes.